src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -476,7 +476,10 @@ static inline ucs_status_t uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface,
         return UCS_OK;
     }
 
-    if (uct_dc_mlx5_iface_dci_can_alloc(iface)) {
+    /* Do not alloc dci if no CQ resources,
+     * otherwise this dci may never be released. */
+    if (uct_dc_mlx5_iface_dci_can_alloc(iface) &&
+        uct_rc_iface_has_tx_resources(&iface->super.super)) {
```

Comment:
maybe do `uct_rc_iface_has_tx_resources` check inside `uct_dc_mlx5_iface_dci_can_alloc`?

Reply:
I'd not do this, because `uct_dc_mlx5_iface_dci_can_alloc` is called in several places, where iface tx resources already checked. For instance `uct_dc_mlx5_ep_flush`. We will check iface resources twice in some cases or will need to make the code a bit more complex to read.

Reply:
yes, these checks go together now, so we can unite them and remove extra ones.

Reply:
this is a simple enough fix, but there is duplicate check: IMHO the macro UCT_DC_MLX5_CHECK_RES should be changed, to check for CQ resources before QP resources (and here we can just have an assertion). Because after this change, we would check for CQ resources twice.

Reply:
Ok, merging this because we need to have a fix at least. We can optimize the resource check later